javascript-bank/account.js

```diff
@@ -0,0 +1,43 @@
+/* exported Account */
+
+function Account(number, holder) {
+  this.number = number;
+  this.holder = holder;
+  this.transactions = [];
+}
+
+Account.prototype.deposit = function (amount) {
+  if (Number.isInteger(amount) && amount > 0) {
+    this.transactions.push(new Transaction('deposit', amount));
+    return true;
+  } else {
+    return false;
+  }
+};
+
+Account.prototype.withdraw = function (amount) {
+  if (Number.isInteger(amount) && amount > 0) {
+    this.transactions.push(new Transaction('withdrawal', amount));
+    return true;
+  } else {
+    return false;
+  }
+};
+
+Account.prototype.getBalance = function () {
+  var total = 0;
+  for (let i = 0; i < this.transactions.length; i++) {
+    if (this.transactions[i].type === 'deposit') {
+      total += this.transactions[i].amount;
+    } else if (this.transactions[i].type === 'withdrawal') {
+      total -= this.transactions[i].amount;
+    }
+  }
+  if (Number.isInteger(total) && total > 0) {
+    return total;
+  } else if (total === 0) {
+    return 0;
+  } else if (total < 0) {
+    return -Math.abs(total);
+  }
```

Comment:
The extra checks are probably not necessary here. If `total` is `0`, `return total;` is the same as `return 0;`. If `total` is negative, then `return total;` is the same as `return -Math.abs(total);`.